src/main/java/data/QueryInformation.java

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableSet;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.TimeZone;
+
+/** Class to hold query information. */
+public class QueryInformation {
+  /** Milliseconds equivalent to 11:59:59.999 PM. */
+  private static final long MILLISECONDS_TO_END_OF_DAY = (24L * 60L * 60L * 1000L) - 1L;
+
+  private final TimeZone timeZone;
+  private final ImmutableSet<String> categories;
+  private final long startTimestamp;
+  private final long endTimestamp;
+  private final String store;
+  private final double minPrice;
+  private final double maxPrice;
+
+  public QueryInformation(String timeZoneId, String categories, String dateRange, String store,
+      String minPrice, String maxPrice)
+      throws ParseException, NumberFormatException, NullPointerException {
+    this.timeZone = TimeZone.getTimeZone(timeZoneId);
+
+    String formattedCategories;
+    try {
+      formattedCategories = formatInput(categories);
+    } catch (NullPointerException exception) {
+      formattedCategories = null;
+    }
+
+    if (!Strings.isNullOrEmpty(formattedCategories)) {
```

Comment:
I'd rearrange this as the following so that the else clause doesn't read as not not null or empty:
```
if (Strings.isNullOrEmpty(...)) {
  this.categories = ImmutableSet.of();
} else {
  ...
}
```
Otherwise